src/delta/delta.c

```diff
@@ -268,11 +268,16 @@ static int enumerate_dir_d(
                 if (!p)
                         return -ENOMEM;
 
+                k = path_extract_filename(p, &bp);
+                if (k < 0)
+                        return k;
+
                 log_debug("Adding to drops: %s %s %s %s %s",
-                          unit, special_glyph(SPECIAL_GLYPH_ARROW_RIGHT), basename(p), special_glyph(SPECIAL_GLYPH_ARROW_RIGHT), p);
-                k = ordered_hashmap_put(h, basename(p), p);
+                          unit, special_glyph(SPECIAL_GLYPH_ARROW_RIGHT), bp, special_glyph(SPECIAL_GLYPH_ARROW_RIGHT), p);
+                k = ordered_hashmap_put(h, bp, p);
                 if (k < 0) {
                         free(p);
+                        free(bp);
```

Comment:
by our current coding style we generally prefer usage of cleanup attributes on local variables, and leave clean-up strictly to them. if you assigned some string elsewhere, so that it is now owned by that something else, use TAKE_PTR() to invalidate the pointer. so that the clenaup attribute's effect is disabled, if you so will.

Reply:
(yes, this pre-existing code doesnt follow that coding style, but another rule here is that if you touch code you also update it to current coding style. or in other words: please also make sure `p` is managed like that.)